### ISSUE TITLE
3w: Create API

### ISSUE THREAD

**Issue by batpad** [IFRCGo/go-api#444]
Relates to #443 - based on the frontend for the 3w data, create API endpoints to:

 - Get all Projects for a Country
 - Edit existing Project (permissions based on Submitting National Society)
 - Add Project (permissions based on National Society user is logged in with?)

cc @szabozoltan69 @guipizzini

**Comment by szabozoltan69** [comment]
Good idea.
Let's see the details, what do we include into a Project!

**Comment by batpad** [comment]
@szabozoltan69 I have created the tickets for what we need on the frontend - [the form](https://github.com/IFRCGo/go-frontend/issues/640) and the [visualization](https://github.com/IFRCGo/go-frontend/issues/641).

We will need an API end-point to receive all the details from the form and Create a New Project, based on the model in the database.

We will probably also need an API end-point to EDIT an existing form.

Also, an end-point where we can pass a Country, and it returns all Projects for that Country, with all the details that are in the model.

I think the rest of the filters, etc. can just be implemented directly on the frontend. So, for a start:

 - Endpoint to receive POST data from client and save a new Project
 - Endpoint to receive PUT data from client and edit an existing Project
 - Endpoint to receive GET request from client with Country parameter, returning all Projects for that country (This will be something like `Project.objects.filter(district__country=<country>)`

We may also need to implement some Permission model so, for example, only a user who belongs to the national society of the `reporting_national_society` is allowed to post a Project as that National Society. I will get more details of the exact permission model here.

@szabozoltan69 let me know if you might be able to take on any of this work.

**Comment by szabozoltan69** [comment]
@batpad I can begin to create endpoints for POST and GET parts, later the others.

**Comment by batpad** [comment]
@szabozoltan69 that sounds fantastic! Let me know if there is anything I can help with - I have started digging into the frontend code and hopefully will be able to start making progress on the form and then the frontend to talk to the APIs.

**Comment by szabozoltan69** [comment]
If I am right, I should do it in the "3w" branch. Can I merge it to develop branch?

**Comment by szabozoltan69** [comment]
Well, I have an obstacle now at creating the first GET endpoint
Object of type 'Currencies' is not JSON serializable

It's true that until now we used this kind of "IntEnum-s":
```
    AFRICA = 0
    AMERICAS = 1
    ASIA_PACIFIC = 2
    EUROPE = 3
    MENA = 4
```
And the Currency abbreviations shows some difficulty.
I tried to make a workaround via "toJson" method, but that did not help.
For me it is fine this CHF, USD symbolization, but let's find out a way with which a api/v2/project/ can be called.
